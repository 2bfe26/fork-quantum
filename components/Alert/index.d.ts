import React from 'react';
import { IconNames } from '../Icon';

export interface AlertProps {
    children: React.ReactNode;
    onClose?: () => {} | React.MouseEventHandler<HTMLButtonElement> | React.TouchEventHandler<HTMLButtonElement>;
    icon?: IconNames | string;
    skin?: 'primary' | 'success' | 'error' | 'neutral' | 'warning';
    theme?: {
        baseFontSize?: number;
        colors?: {};
        spacing?: {};
        components?: {
            alert?: {};
            button?: {};
        };
    };
}

export default class Alert extends React.Component<AlertProps> {}
