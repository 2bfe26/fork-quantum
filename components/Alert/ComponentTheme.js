import Colors from '../Colors';

const ComponentTheme = {
  success: {
    backgroundColor: Colors.SUCCESS['500'],
    color: Colors.WHITE,
    icon: 'check',
  },
  warning: {
    backgroundColor: Colors.WARNING['500'],
    color: Colors.WHITE,
    icon: 'warning',
  },
  error: {
    backgroundColor: Colors.DANGER['500'],
    color: Colors.WHITE,
    icon: 'do_not_disturb_alt',
  },
  info: {
    backgroundColor: Colors.INFO['500'],
    color: Colors.WHITE,
    icon: 'info_outline',
  },
};

export default ComponentTheme;
