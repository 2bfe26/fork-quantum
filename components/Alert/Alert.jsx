import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';
import React, { Component } from 'react';
import Icon from '../Icon';
import Colors from '../Colors';

import GlobalTheme from '../../theme';
import ComponentTheme from './ComponentTheme';

const sharedStyles = `
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
`;

const AlertWrapper = styled.div`
  background-color: ${props => ComponentTheme[props.type].backgroundColor};
  border-radius: 2px;
  color: ${props => ComponentTheme[props.type].color};
  padding: 12px;
  position: relative;
  ${GlobalTheme.mixins.shadow(2)}
`;

const AlertIcon = styled(Icon)`
  ${sharedStyles}
  left: 12px;
`;

const AlertMessage = styled.div`
  display: inline-block;
  margin-left: 35px;
  margin-right: 35px;
`;

const AlertTitle = styled.strong`
  display: block;
`;

const AlertClose = styled(Icon)`
  && {
    ${sharedStyles}
    background-color: ${GlobalTheme.mixins.hexToRgba(Colors.WHITE, 0.5)};
    border-radius: 50%;
    cursor: pointer;
    font-size: 12px;
    height: 17px;
    line-height: 17px;
    right: 12px;
    text-align: center;
    width: 17px;
  }
`;

class Alert extends Component {
  constructor(props) {
    super(props);

    this.state = {
      show: this.props.show,
    };

    if (this.props.duration) {
      this.handleAutoDestruction();
    }
  }

  hideModal = () => {
    this.setState({ show: false }, this.props.onClose ? this.props.onClose() : null);
  }

  handleAutoDestruction = () => {
    setTimeout(() => {
      this.hideModal();
    }, this.props.duration * 1000);
  }

  render() {
    const { type, message: { title, text } } = this.props;

    return this.state.show ? (
      <ThemeProvider theme={GlobalTheme}>
        <AlertWrapper type={type}>
          <AlertIcon name={ComponentTheme[type].icon} />

          <AlertMessage>
            <AlertTitle>
              {title}
            </AlertTitle>

            {text}
          </AlertMessage>

          <AlertClose title="Fechar" skin={ComponentTheme[type].backgroundColor} name="close" onClick={this.hideModal} />
        </AlertWrapper>
      </ThemeProvider>
    ) : null;
  }
}

AlertClose.displayName = 'AlertClose';
Alert.defaultProps = {
  type: 'info',
  message: {
    title: 'Simple',
    text: 'This is a dummy',
  },
  duration: 0,
  show: true,
  onClose: () => {},
};

Alert.propTypes = {
  /** alert type */
  type: PropTypes.oneOf([
    'info',
    'success',
    'warning',
    'error',
  ]),
  /** time in seconds that Alert is visible */
  duration: PropTypes.number,
  /** it says to component if it should appear or not */
  show: PropTypes.bool,
  /** object with message details */
  message: PropTypes.shape({
    /** message title */
    title: PropTypes.string,
    /** message text */
    text: PropTypes.string,
  }),
  /** it says if the title should stay in the same line of the message itself */
  /** callback function */
  onClose: PropTypes.func,
};

export default Alert;
