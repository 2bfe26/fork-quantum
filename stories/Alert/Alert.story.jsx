import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  AutoExample,
  Tab,
  Example,
  Title,
} from '@catho-private/quantum-storybook-ui';

import Alert from '../../components/Alert';

const exampleTab = (
  <Tab title="Example">
    <p>You can specify a duration in seconds of the alert element.</p>
    <Title>With 5 seconds of duration</Title>
    <Example
      component={
        <Alert
          show
          type="error"
          duration={5}
          message={{
            title: 'Simple',
            text: 'This is error alert',
          }}
        />
      }
    />
    <p>We have 4 type of alerts:</p>
    <ul>
      <li>Info</li>
      <li>Success</li>
      <li>Warning</li>
      <li>Danger</li>
    </ul>
    <Title>Info</Title>
    <Example
      component={
        <Alert
          show
          type="info"
          message={{
            title: 'Info',
            text: 'This is info alert',
          }}
        />
      }
    />
    <Title>Success</Title>
    <Example
      component={
        <Alert
          show
          type="success"
          message={{
            title: 'Success',
            text: 'This is success alert',
          }}
        />
      }
    />
    <Title>Warning</Title>
    <Example
      component={
        <Alert
          show
          type="warning"
          message={{
            title: 'Warning',
            text: 'This is warning alert',
          }}
        />
      }
    />
    <Title>Error</Title>
    <Example
      component={
        <Alert
          show
          type="error"
          message={{
            title: 'Error',
            text: 'This is error alert',
          }}
        />
      }
    />
  </Tab>
);

storiesOf('Alerts', module).add('Basic', () => (
  <AutoExample component={Alert} additionalTabs={exampleTab} />
));
