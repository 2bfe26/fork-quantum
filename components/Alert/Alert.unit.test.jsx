import React from 'react';
import renderer from 'react-test-renderer';
import Alert from './Alert';

describe('Alert component', () => {
  it('Should match the snapshot of a simple alert', () => {
    const simpleAlert = <Alert onClose={() => {}}>Sample alert</Alert>;
    expect(renderer.create(simpleAlert).toJSON()).toMatchSnapshot();
  });

  describe('When you set a alert custom icon', () => {
    it('Should match the snapshot with an icon', () => {
      const simpleAlert = (
        <Alert onClose={() => {}} icon="info">
          Sample alert
        </Alert>
      );
      expect(renderer.create(simpleAlert).toJSON()).toMatchSnapshot();
    });
  });

  describe('When you set a different skin', () => {
    it('Should match a skin snapshot', () => {
      const neutral = (
        <Alert onClose={() => {}} skin="neutral">
          blue
        </Alert>
      );
      const primary = (
        <Alert onClose={() => {}} skin="primary">
          blue
        </Alert>
      );
      const success = (
        <Alert onClose={() => {}} skin="success">
          success
        </Alert>
      );
      const warning = (
        <Alert onClose={() => {}} skin="warning">
          warning
        </Alert>
      );
      const error = (
        <Alert onClose={() => {}} skin="error">
          error
        </Alert>
      );
      expect(renderer.create(neutral).toJSON()).toMatchSnapshot();
      expect(renderer.create(primary).toJSON()).toMatchSnapshot();
      expect(renderer.create(success).toJSON()).toMatchSnapshot();
      expect(renderer.create(warning).toJSON()).toMatchSnapshot();
      expect(renderer.create(error).toJSON()).toMatchSnapshot();
    });
  });
});
