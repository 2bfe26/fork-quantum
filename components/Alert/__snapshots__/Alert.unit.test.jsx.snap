// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component Should match the snapshot of a simple alert 1`] = `
.c7 {
  pointer-events: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c5 *:nth-child(2) {
  margin-left: 5px;
}

.c5 .c6 {
  font-size: 24px;
}

.c5:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c5:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c5:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c2 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  padding: 11px 16px;
  color: undefined;
  background-color: #e5edfc;
  border: 1.5px solid #1355d0;
}

.c0 .c1 .c8 {
  color: #1355d0;
  margin-right: 16px;
}

.c0 .c1 .c3 {
  color: #1355d0;
}

.c4 {
  border-radius: 50%;
  border: none;
  color: #999999;
  width: 40px;
  background-color: transparent;
  box-shadow: none;
  height: auto;
  width: auto;
  margin: 0 0 0 16px;
  padding: 0;
  opacity: 0.8;
  -webkit-transition: opacity 0.4s ease;
  transition: opacity 0.4s ease;
}

.c4:hover,
.c4:focus {
  box-shadow: none;
  background-color: rgba(204,204,204,0.4);
  color: #4c4c4c;
}

.c4:active {
  box-shadow: none;
  background-color: rgba(128,128,128,0.5);
  color: #4c4c4c;
}

.c4:hover {
  background: none;
  opacity: 1;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    onClose={[Function]}
  >
    <span>
      Sample alert
    </span>
    <button
      className="c3 c4 c5"
      disabled={false}
      onClick={[Function]}
      size="medium"
      type="button"
    >
      <span
        aria-hidden="true"
        className="material-icons MuiIcon-root-1 c6 c7"
        size="medium"
        style={
          Object {
            "color": "",
          }
        }
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`Alert component When you set a alert custom icon Should match the snapshot with an icon 1`] = `
.c8 {
  pointer-events: none;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c6 *:nth-child(2) {
  margin-left: 5px;
}

.c6 .c7 {
  font-size: 24px;
}

.c6:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c6:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c6:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c2 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  padding: 11px 16px;
  color: undefined;
  background-color: #e5edfc;
  border: 1.5px solid #1355d0;
}

.c0 .c1 .c3 {
  color: #1355d0;
  margin-right: 16px;
}

.c0 .c1 .c4 {
  color: #1355d0;
}

.c5 {
  border-radius: 50%;
  border: none;
  color: #999999;
  width: 40px;
  background-color: transparent;
  box-shadow: none;
  height: auto;
  width: auto;
  margin: 0 0 0 16px;
  padding: 0;
  opacity: 0.8;
  -webkit-transition: opacity 0.4s ease;
  transition: opacity 0.4s ease;
}

.c5:hover,
.c5:focus {
  box-shadow: none;
  background-color: rgba(204,204,204,0.4);
  color: #4c4c4c;
}

.c5:active {
  box-shadow: none;
  background-color: rgba(128,128,128,0.5);
  color: #4c4c4c;
}

.c5:hover {
  background: none;
  opacity: 1;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    onClose={[Function]}
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-1 c3 "
      style={
        Object {
          "color": "",
        }
      }
    >
      info
    </span>
    <span>
      Sample alert
    </span>
    <button
      className="c4 c5 c6"
      disabled={false}
      onClick={[Function]}
      size="medium"
      type="button"
    >
      <span
        aria-hidden="true"
        className="material-icons MuiIcon-root-1 c7 c8"
        size="medium"
        style={
          Object {
            "color": "",
          }
        }
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`Alert component When you set a different skin Should match a skin snapshot 1`] = `
.c7 {
  pointer-events: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c5 *:nth-child(2) {
  margin-left: 5px;
}

.c5 .c6 {
  font-size: 24px;
}

.c5:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c5:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c5:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c2 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  padding: 11px 16px;
  color: undefined;
  background-color: #e5edfc;
  border: 1.5px solid #1355d0;
}

.c0 .c1 .c8 {
  color: #1355d0;
  margin-right: 16px;
}

.c0 .c1 .c3 {
  color: #1355d0;
}

.c4 {
  border-radius: 50%;
  border: none;
  color: #999999;
  width: 40px;
  background-color: transparent;
  box-shadow: none;
  height: auto;
  width: auto;
  margin: 0 0 0 16px;
  padding: 0;
  opacity: 0.8;
  -webkit-transition: opacity 0.4s ease;
  transition: opacity 0.4s ease;
}

.c4:hover,
.c4:focus {
  box-shadow: none;
  background-color: rgba(204,204,204,0.4);
  color: #4c4c4c;
}

.c4:active {
  box-shadow: none;
  background-color: rgba(128,128,128,0.5);
  color: #4c4c4c;
}

.c4:hover {
  background: none;
  opacity: 1;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    onClose={[Function]}
  >
    <span>
      blue
    </span>
    <button
      className="c3 c4 c5"
      disabled={false}
      onClick={[Function]}
      size="medium"
      type="button"
    >
      <span
        aria-hidden="true"
        className="material-icons MuiIcon-root-1 c6 c7"
        size="medium"
        style={
          Object {
            "color": "",
          }
        }
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`Alert component When you set a different skin Should match a skin snapshot 2`] = `
.c7 {
  pointer-events: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c5 *:nth-child(2) {
  margin-left: 5px;
}

.c5 .c6 {
  font-size: 24px;
}

.c5:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c5:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c5:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c2 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 .c1 .c9 {
  color: #1355d0;
  margin-right: 16px;
}

.c8 .c1 .c3 {
  color: #1355d0;
}

.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  padding: 11px 16px;
  color: #3c6510;
  background-color: #edfadf;
  border: 1.5px solid #7ed321;
}

.c0 .c1 .c9 {
  color: #7ed321;
  margin-right: 16px;
}

.c0 .c1 .c3 {
  color: #7ed321;
}

.c4 {
  border-radius: 50%;
  border: none;
  color: #999999;
  width: 40px;
  background-color: transparent;
  box-shadow: none;
  height: auto;
  width: auto;
  margin: 0 0 0 16px;
  padding: 0;
  opacity: 0.8;
  -webkit-transition: opacity 0.4s ease;
  transition: opacity 0.4s ease;
}

.c4:hover,
.c4:focus {
  box-shadow: none;
  background-color: rgba(204,204,204,0.4);
  color: #4c4c4c;
}

.c4:active {
  box-shadow: none;
  background-color: rgba(128,128,128,0.5);
  color: #4c4c4c;
}

.c4:hover {
  background: none;
  opacity: 1;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    onClose={[Function]}
  >
    <span>
      success
    </span>
    <button
      className="c3 c4 c5"
      disabled={false}
      onClick={[Function]}
      size="medium"
      type="button"
    >
      <span
        aria-hidden="true"
        className="material-icons MuiIcon-root-1 c6 c7"
        size="medium"
        style={
          Object {
            "color": "",
          }
        }
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`Alert component When you set a different skin Should match a skin snapshot 3`] = `
.c7 {
  pointer-events: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c5 *:nth-child(2) {
  margin-left: 5px;
}

.c5 .c6 {
  font-size: 24px;
}

.c5:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c5:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c5:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c2 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 .c1 .c10 {
  color: #1355d0;
  margin-right: 16px;
}

.c8 .c1 .c3 {
  color: #1355d0;
}

.c9 .c1 .c10 {
  color: #7ed321;
  margin-right: 16px;
}

.c9 .c1 .c3 {
  color: #7ed321;
}

.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  padding: 11px 16px;
  color: #a36300;
  background-color: #ffefd6;
  border: 1.5px solid #f09100;
}

.c0 .c1 .c10 {
  color: #f09100;
  margin-right: 16px;
}

.c0 .c1 .c3 {
  color: #f09100;
}

.c4 {
  border-radius: 50%;
  border: none;
  color: #999999;
  width: 40px;
  background-color: transparent;
  box-shadow: none;
  height: auto;
  width: auto;
  margin: 0 0 0 16px;
  padding: 0;
  opacity: 0.8;
  -webkit-transition: opacity 0.4s ease;
  transition: opacity 0.4s ease;
}

.c4:hover,
.c4:focus {
  box-shadow: none;
  background-color: rgba(204,204,204,0.4);
  color: #4c4c4c;
}

.c4:active {
  box-shadow: none;
  background-color: rgba(128,128,128,0.5);
  color: #4c4c4c;
}

.c4:hover {
  background: none;
  opacity: 1;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    onClose={[Function]}
  >
    <span>
      warning
    </span>
    <button
      className="c3 c4 c5"
      disabled={false}
      onClick={[Function]}
      size="medium"
      type="button"
    >
      <span
        aria-hidden="true"
        className="material-icons MuiIcon-root-1 c6 c7"
        size="medium"
        style={
          Object {
            "color": "",
          }
        }
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`Alert component When you set a different skin Should match a skin snapshot 4`] = `
.c7 {
  pointer-events: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c5 *:nth-child(2) {
  margin-left: 5px;
}

.c5 .c6 {
  font-size: 24px;
}

.c5:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c5:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c5:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c2 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 .c1 .c11 {
  color: #1355d0;
  margin-right: 16px;
}

.c8 .c1 .c3 {
  color: #1355d0;
}

.c9 .c1 .c11 {
  color: #7ed321;
  margin-right: 16px;
}

.c9 .c1 .c3 {
  color: #7ed321;
}

.c10 .c1 .c11 {
  color: #f09100;
  margin-right: 16px;
}

.c10 .c1 .c3 {
  color: #f09100;
}

.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  padding: 11px 16px;
  color: #ac001a;
  background-color: #ffe5e9;
  border: 1.5px solid #ff2d4d;
}

.c0 .c1 .c11 {
  color: #ff2d4d;
  margin-right: 16px;
}

.c0 .c1 .c3 {
  color: #ff2d4d;
}

.c4 {
  border-radius: 50%;
  border: none;
  color: #999999;
  width: 40px;
  background-color: transparent;
  box-shadow: none;
  height: auto;
  width: auto;
  margin: 0 0 0 16px;
  padding: 0;
  opacity: 0.8;
  -webkit-transition: opacity 0.4s ease;
  transition: opacity 0.4s ease;
}

.c4:hover,
.c4:focus {
  box-shadow: none;
  background-color: rgba(204,204,204,0.4);
  color: #4c4c4c;
}

.c4:active {
  box-shadow: none;
  background-color: rgba(128,128,128,0.5);
  color: #4c4c4c;
}

.c4:hover {
  background: none;
  opacity: 1;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    onClose={[Function]}
  >
    <span>
      error
    </span>
    <button
      className="c3 c4 c5"
      disabled={false}
      onClick={[Function]}
      size="medium"
      type="button"
    >
      <span
        aria-hidden="true"
        className="material-icons MuiIcon-root-1 c6 c7"
        size="medium"
        style={
          Object {
            "color": "",
          }
        }
      >
        close
      </span>
    </button>
  </div>
</div>
`;
